src/raven.js

```diff
@@ -843,6 +843,38 @@ function makeRequest(opts) {
     img.src = src;
 }
 
+function makeXhrRequest(opts) {
+    var request = new XMLHttpRequest();
+    request.onreadystatechange = function (e) {
+        if (request.readyState !== 4) {
+            return;
+        }
+
+        if (request.status === 200) {
+            if (opts.onSuccess) {
+                opts.onSuccess();
+            }
+        } else {
+            if (opts.onError) {
+                opts.onError();
+            }
+        }
+    };
+
+    // NOTE: auth is intentionally sent as part of query string (NOT as custom
+    //       HTTP header) so as to avoid preflight CORS requests
+    request.open('POST', opts.url + '?' + urlencode(opts.auth));
+    request.send(JSON.stringify(opts.data));
+}
+
+function makeRequest(opts) {
+    var hasCORS =
+        'withCredentials' in new XMLHttpRequest() ||
+        typeof XDomainRequest !== 'undefined';
+
+    return (hasCORS ? makeXhrRequest : makeImageRequest)(opts);
```

Comment:
I agree. So we'll have to implement the stuff to make sure we trim our data packet before sending to make an attempt.

Otherwise, we'll just get a 413 on the response and Sentry won't log it anyways.

So not sure the best solution here.